net/dispatcher.go

```
@@ -221,13 +221,16 @@ func BlockListOption(blockList []string) DispatcherOption {
 
 // InsecureSkipVerifyOption allow self-signed certificates
 // to be used if set to true but is susceptible to Man In The Middle attacks.
-func InsecureSkipVerifyOption(insecureSkipVerify bool) DispatcherOption {
+func InsecureSkipVerifyOption(insecureSkipVerify bool, licenser license.Licenser, caCertTLSConfig *tls.Config) DispatcherOption {
 	return func(d *Dispatcher) error {
-		if insecureSkipVerify {
+		switch {
+		case insecureSkipVerify:
 			d.transport.TLSClientConfig = &tls.Config{
 				InsecureSkipVerify: true,
 			}
-		} else {
+		case licenser.CustomCertificateAuthority() && caCertTLSConfig != nil:
+			d.transport.TLSClientConfig = caCertTLSConfig
+		default:
```

Adding this functionality here pollutes this option because this change makes it do more than one thing. It should be a new option, and when applying the options, we should return an error if both are supplied.

---

I'll rename the option